src/random/pcg32.cr

```diff
@@ -38,20 +38,22 @@ class Random::PCG32
 
   @state : UInt64
   @inc : UInt64
+  @initstate : UInt64
+  @initseq : UInt64
 
   def self.new
     new(Random::Secure.rand(UInt64::MIN..UInt64::MAX), Random::Secure.rand(UInt64::MIN..UInt64::MAX))
   end
 
-  def initialize(initstate : UInt64, initseq = 0_u64)
+  def initialize(@initstate : UInt64, @initseq = 0_u64)
     # initialize to zeros to prevent compiler complains
     @state = 0_u64
     @inc = 0_u64
-    new_seed(initstate, initseq)
+    new_seed(@initstate, @initseq)
   end
 
-  def new_seed
-    new_seed(Random::Secure.rand(UInt64::MIN..UInt64::MAX), Random::Secure.rand(UInt64::MIN..UInt64::MAX))
+  def init_seed
```

Comment:
I don't think we need this. We should generate the seed using a random, then use that seed to initialize another random. Then we don't need to ask the seed from the random (we generate it ourselves for the case of `random`).

Reply:
That's a good idea...I'll update